### PR TITLE
Add regression test for the mark yellow background and markup persistence issue

### PR DESCRIPTION
**Pull request by fullofcaffeine**
Related to: https://github.com/WordPress/gutenberg/pull/35516

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/rich-text.test.js [hunk 1]
```diff
@@ -548,4 +549,51 @@ describe( 'RichText', () => {
 		// Expect: <strong>1</strong>-<em>2</em>
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	// Regression test for https://github.com/WordPress/gutenberg/pull/35516.
+	it.only( 'should properly handle pre-<mark> `text-color` markup', async () => {
```
Comment (fullofcaffeine) [inline packages/e2e-tests/specs/editor/various/rich-text.test.js:554]: We should probably come up with a better description here.

File: packages/e2e-tests/specs/editor/various/rich-text.test.js [hunk 2]
```diff
@@ -548,4 +549,51 @@ describe( 'RichText', () => {
...
+			editorCustomColorText,
+		] = await page.evaluate( () =>
+			Array.from(
+				document.querySelectorAll(
+					'p[aria-label="Paragraph block"] > *'
+				)
+			).map( ( el ) => ( {
+				tag: el.localName,
+				style: el.getAttribute( 'style' ),
+				class: el.getAttribute( 'class' ),
```
Comment (fullofcaffeine) [inline packages/e2e-tests/specs/editor/various/rich-text.test.js:583]: We can probably remove this attribute, as it seems we won't be checking against it.

**Comment by chad1008** [comment]
Due to access level on the repo, I needed to submit through my own fork, so I have a new PR with the rest of the regression test here: https://github.com/WordPress/gutenberg/pull/35847

File: packages/e2e-tests/specs/editor/various/rich-text.test.js (hunk 2 above)
Comment (chad1008) [reply packages/e2e-tests/specs/editor/various/rich-text.test.js:583]: Agreed, I've removed in my PR (see below)

File: packages/e2e-tests/specs/editor/various/rich-text.test.js (hunk 1 above)
Comment (chad1008) [reply packages/e2e-tests/specs/editor/various/rich-text.test.js:554]: Agree! I've introduced a new description in the PR below.

**Pull request closed without merging**